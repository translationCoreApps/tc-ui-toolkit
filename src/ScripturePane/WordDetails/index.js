import React from 'react';
import PropTypes from 'prop-types';
// helpers
import * as lexiconHelpers from '../helpers/lexiconHelpers';

class WordDetails extends React.Component {
  render() {
    const {
      word:{
      lemma, morph, strong
    },
    translate
  } = this.props;
    const { lexiconData } = this.props;
    const entryId = lexiconHelpers.lexiconEntryIdFromStrongs(strong);
    const lexiconId = lexiconHelpers.lexiconIdFromStrongs(strong);
    let lexicon;
    if (lexiconData[lexiconId] && lexiconData[lexiconId][entryId]) {
      lexicon = lexiconData[lexiconId][entryId].long;
    }

    return (
      <div style={{ margin: '-10px 10px -20px', maxWidth: '400px' }}>
        <span><strong>{translate('lemma')}</strong> {lemma}</span><br/>
        <span><strong>{translate('morph')}</strong> {morph}</span><br/>
        <span><strong>{translate('strongs')}</strong> {strong}</span><br/>
        <span><strong>{translate('lexicon')}</strong> {lexicon}</span><br/>
      </div>
    );
  }
}

WordDetails.propTypes = {
  word: PropTypes.object.isRequired,
  lexiconData: PropTypes.object.isRequired
};

export default WordDetails;
