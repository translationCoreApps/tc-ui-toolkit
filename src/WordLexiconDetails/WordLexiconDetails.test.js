import React from 'react';
import WordDetails from './WordLexiconDetails';
import renderer from 'react-test-renderer';

describe('WordDetails', () => {
  it('renders Greek', () => {
    const wrapper = renderer.create(
      <WordDetails
        lexiconData={{}}
        wordObject={{
          lemma: 'lemma',
          morph: 'Gr,NS,,,,GMPF',
          strong: 'strong'
        }}
        translate={k=>k}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders Hebrew', () => {
    const wrapper = renderer.create(
      <WordDetails
        lexiconData={{}}
        wordObject={{
          lemma: 'lemma',
          morph: 'He,R:Ncfsa',
          strong: 'b:H7225'
        }}
        translate={k=>k}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
