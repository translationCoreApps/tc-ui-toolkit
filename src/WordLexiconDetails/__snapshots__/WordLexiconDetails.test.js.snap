// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordDetails renders Greek 1`] = `
<div
  style={
    Object {
      "margin": "-10px 10px -20px",
      "maxWidth": "400px",
    }
  }
>
  <span>
    <strong>
      lemma
    </strong>
     
    lemma
  </span>
  <br />
  <span>
    <strong>
      morphology
    </strong>
     
    noun, substantive_adj, genitive, masculine, plural, F
  </span>
  <br />
  <span>
    <strong>
      strongs
    </strong>
     
    strong
  </span>
  <br />
  <span>
    <strong>
      lexicon
    </strong>
     
  </span>
  <br />
</div>
`;

exports[`WordDetails renders Hebrew 1`] = `
<div
  style={
    Object {
      "margin": "-10px 10px -20px",
      "maxWidth": "400px",
    }
  }
>
  <span>
    <strong>
      lemma
    </strong>
     
    lemma
  </span>
  <br />
  <span>
    <strong>
      morphology
    </strong>
     
    preposition: noun, common, feminine, singular, absolute
  </span>
  <br />
  <span>
    <strong>
      strongs
    </strong>
     
    b:H7225
  </span>
  <br />
  <span>
    <strong>
      lexicon
    </strong>
     
  </span>
  <br />
</div>
`;
